metrics.go

```diff
@@ -39,7 +40,11 @@ func (mc MetricsCollector) Describe(ch chan<- *prometheus.Desc) {
 
 // Collect implements the prometheus.Collector interface.
 func (mc MetricsCollector) collectMetrics(ch chan<- prometheus.Metric) {
-	pending := recentlyPendingCSRs(mc.cacheIndexer)
+	csrs := []*certificatesv1beta1.CertificateSigningRequest{}
+	for _, item := range mc.cacheIndexer.List() {
+		csrs = append(csrs, item.(*certificatesv1beta1.CertificateSigningRequest))
+	}
```

Comment:
The fact we had to move this twice, definitely suggests we should be making the tests bend to the code rather than the other way around